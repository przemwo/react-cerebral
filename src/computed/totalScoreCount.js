import { compute } from 'cerebral';
import { state } from 'cerebral/tags';

function totalScoreCount (playersKeys, get) {
  return (playersKeys || []).reduce(
    (currentCount, playerKey) => {
      return currentCount + get(state`players.data.${playerKey}.score`);
    },
    0
  );
}

export default compute(
  state`players.data.*`,
  totalScoreCount
);
