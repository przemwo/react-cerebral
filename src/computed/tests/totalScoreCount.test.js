import { runCompute } from 'cerebral/test';

import totalScoreCount from '../totalScoreCount';

describe('totalScoreCount() computed', () => {
  it(`returns sum of players' scores`, () => {
    const fakeState = {
      players: {
        data: {
          '443fd35f': {
            name: 'player1',
            score: 10
          },
          '187457c2': {
            name: 'player2',
            score: 20
          },
          '5a89a1d9': {
            name: 'player3',
            score: -5
          }
        }
      }
    };
    const result = runCompute(totalScoreCount, { state: fakeState });

    expect(result).toMatchSnapshot();
  });
});
