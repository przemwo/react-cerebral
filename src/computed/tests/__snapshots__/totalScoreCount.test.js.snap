exports[`totalScoreCount() computed returns sum of players' scores 1`] = `25`;
